src/main/java/org/opensearch/security/support/HeaderHelper.java

```diff
@@ -57,15 +57,8 @@ public static String getSafeFromHeader(final ThreadContext context, final String
             return null;
         }
 
-        String headerValue = null;
-        	
-        Map<String, String> headers = context.getHeaders();
-        if (!headers.containsKey(headerName) || (headerValue = headers.get(headerName)) == null) {
-            return null;
-        }
-
         if (isInterClusterRequest(context) || isTrustedClusterRequest(context) || isDirectRequest(context)) {
-            return headerValue;
+            return context.getHeader(headerName);
         }
 
```

Comment:
It is surprising that this small change makes a sizeable difference.

Reply:
> Changes are good - can you share the experiment results?

* HeaderHelper.deserializeSafeFromHeader ‘s overall CPU usage dropped to 0.52% from 5%-12.5% seen across multiple CPU profiles obtained for prior runs.
* Search Latency saw a drop of ~75%, Search Throughput increased by ~175%
* Indexing Latency dropped by  70%, throughput increased by more than ~50%.